src/Util/FisherYates/Interface.dfy

```diff
@@ -0,0 +1,16 @@
+/*******************************************************************************
+ *  Copyright by the contributors to the Dafny Project
+ *  SPDX-License-Identifier: MIT
+ *******************************************************************************/
+
+module FisherYates.Interface {
+  import Uniform
+
+  trait {:termination false} Trait extends Uniform.Interface.Trait {
+
+    method Shuffle<T(!new)>(a: array<T>)
```

Comment:
Depending on what will be shuffled using this, thee `(!new)`, may be undesirable.

Reply:
Good catch. Not sure why I added it, it doesn't seem to be needed. Perhaps a relict of experiments in the previous PR. Removing it.